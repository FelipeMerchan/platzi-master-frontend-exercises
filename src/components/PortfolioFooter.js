import React from 'react'

import PortfolioMenu from './PortfolioMenu'
import './styles/PortfolioFooter.css'

function PortfolioFooter(props) {
  return (
    <footer className="PortfolioFooter" id="footer">
      <PortfolioMenu onClick={props.onClick} dataset="footer"/>
      <div>
        <span>Copyright 2010 <strong>- Felipe Merchan</strong> - Todos los derechos reservados</span>
      </div>
    </footer>
  )
}

export default PortfolioFooter