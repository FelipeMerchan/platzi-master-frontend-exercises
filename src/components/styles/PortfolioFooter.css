.PortfolioFooter {
  display: flex;
  align-items: center;
  justify-content: space-between;
  color: #afafaf;
}

.PortfolioFooter strong {
  color: #c5345c;
}

@media screen and (max-width: 768px) {
  .PortfolioFooter {
    display: block;
    padding: var(--space-2) 0;
  }
}