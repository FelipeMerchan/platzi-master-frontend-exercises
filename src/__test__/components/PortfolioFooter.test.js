import React from 'react'
import { shallow, mount } from 'enzyme'
import { create } from 'react-test-renderer'
import '../../setupTest'
import PortfolioFooter from '../../components/PortfolioFooter'

describe('Probar el componente <PortfolioFooter />', () => {
  const portfolioFooter = shallow(<PortfolioFooter />)
  test('Render del componente PortfolioFooter', () => {
    expect(portfolioFooter.length).toEqual(1)
  })
  test('Render del Copyright', () => {
    expect(portfolioFooter.exists('span')).toBeTruthy()
  })
})

describe('PortfolioFooter Snapshot', () => {
  const portfolioFooter = create(<PortfolioFooter />)
  test('Comprobar la UI del componente PortfolioFooter', () => {
    expect(portfolioFooter.toJSON()).toMatchSnapshot()
  })
})