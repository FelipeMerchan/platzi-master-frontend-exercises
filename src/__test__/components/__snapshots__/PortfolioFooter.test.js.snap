// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortfolioFooter Snapshot Comprobar la UI del componente PortfolioFooter 1`] = `
<footer
  className="PortfolioFooter"
  id="footer"
>
  <nav
    className="PortfolioMenu"
  >
    <ul>
      <li
        className="PortfolioMenu__item"
      >
        <a
          data-button="footer"
          href="#home"
          id="home"
          onClick={[Function]}
          role="button"
        >
          inicio
        </a>
      </li>
      <li
        className="PortfolioMenu__item"
      >
        <a
          data-button="footer"
          href="#aboutMe"
          id="aboutMe"
          onClick={[Function]}
          role="button"
        >
          acerca
        </a>
      </li>
      <li
        className="PortfolioMenu__item"
      >
        <a
          data-button="footer"
          href="#portfolio"
          id="portfolio"
          onClick={[Function]}
          role="button"
        >
          portafolio
        </a>
      </li>
      <li
        className="PortfolioMenu__item"
      >
        <a
          data-button="footer"
          href="#contact"
          id="contact"
          onClick={[Function]}
          role="button"
        >
          contacto
        </a>
      </li>
    </ul>
  </nav>
  <div>
    <span>
      Copyright 2010 
      <strong>
        - Felipe Merchan
      </strong>
       - Todos los derechos reservados
    </span>
  </div>
</footer>
`;
